ansible/roles/etcd/tasks/etcd2-configure.yml

```diff
@@ -11,10 +11,3 @@
 - name: CoreOS | Write etcd config file
   template: src=etcd.conf.j2 dest=/etc/etcd/etcd.conf
   notify: restart etcd2
```

Comment:
``restart etcd`` handler exists to make the deployment script idempotent.

* When running the role on vanilla cluster, etcd is not installed (thus not running). Running ``Start etcd`` task sets ``etcd_started.changed`` to ``true``. Before the ``etcd`` role ends, handlers are checked. As ``etcd_started.changed`` is not ``false``, ``restart etcd`` handler is not called.
* When running the role on a cluster with etcd installed and running, ``Start etcd`` task sets ``etcd_started.changed`` to ``false``. Thus, the ``restart etcd`` handler is called.

To run the ``restart etcd`` handler at all, ``Start etcd`` has to be run at the first place. So if you run ``install`` and/or ``etcd-configure`` without ``etcd-start``, the handler is not usable. At the same time you need to restart the etcd service if you change configuration file or update to newer version of etcd. So the ``restart etcd`` must be independent of ``Start etcd`` task.

Thus, instead of keeping ``restart etcd`` conditional, I suggest:

* make the ``restart etcd`` handler unconditional
* introduce new ``etcd_modified`` variable in ``defaults/main.yaml`` set to ``false`` by default
* if any of ``install-etcd`` and ``configure-etcd`` (and its equivalents) gets invoked, ``etcd_modified`` variable gets set to ``true``
* at the end of ``tasks/main.yml`` we will test if ``etcd_modified`` is ``true`` and ``etcd_started.changed`` is ``false``. If so, ``restart etcd`` gets notified and called.